jib-core/src/main/java/com/google/cloud/tools/jib/builder/BuildConfiguration.java

```diff
@@ -96,14 +97,14 @@ public Builder setMainClass(@Nullable String mainClass) {
       return this;
     }
 
-    public Builder setJavaArguments(@Nullable List<String> javaArguments) {
+    public Builder setJavaArguments(@Nullable ImmutableList<String> javaArguments) {
```

Comment:
It's a little strange to use Immutable list in the API to this. Especially if in the future we don't export Guava outside of our library.

Reply:
Could we not just accept List<String> and then make a copy?

Reply:
I think the idea was to avoid having to do this null-element validation in jib-core, but I'm not sure. @coollog?

Reply:
I guess our contract to these calls should be non-null, non-empty values, but we should probably enforce it as this level?